src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -119,49 +122,43 @@
 
     private SqlFedAuthToken fedAuthToken = null;
 
-    static class Sha1HashKey {
-        private byte[] bytes;
+    static class CityHash128Key {
```

Comment:
Simply performance.  On a local benchmark I measured, for 100,000 hashes:

SHA1 Hash Total ns: 124926909 (~125ms)

CityHashCrc128 Total ns: 23457110 (~23.5ms)

CitiHash128 passes the [smhasher](https://github.com/rurban/smhasher/blob/master/doc/City128) with no collisions, which is what we're really after as much as performance.